File: Algorithms/119. Pascal's Triangle II/swift/solution.dp2.swift
class Solution {
    func getRow(_ rowIndex: Int) -> [Int] {
        var ans: [Int] = Array(repeating: 1, count: rowIndex + 1)
        guard rowIndex > 0 else {
            return ans
        }
        for i in 1..<rowIndex {
            for j in stride(from: i, to: 0, by: -1) {
                ans[j] += ans[j - 1]
            }
        }
        return ans
    }
}

File: Algorithms/119. Pascal's Triangle II/python/solution.dp2.py
class Solution:
    def getRow(self, rowIndex: int) -> list[int]:
        ans = [1] * (rowIndex + 1)
        for i in range(1, rowIndex):
            for j in range(i, 0, -1):
                ans[j] += ans[j - 1]
        return ans
